python/pyspark/shuffle.py

```diff
@@ -367,32 +372,13 @@ def iteritems(self):
 
     def _external_items(self):
         """ Return all partitioned items as iterator """
-        assert not self.data
         if any(self.pdata):
             self._spill()
-        hard_limit = self._next_limit()
+        self.pdata = []
```

Comment:
It looks like the old code didn't clear `pdata` after spilling, but I'm guessing that wasn't a problem because we didn't iterate multiple times before?